posthog/hogql/database/database.py

```diff
@@ -225,8 +234,17 @@ def create_hogql_database(
 
     team = team_arg or Team.objects.get(pk=team_id)
     modifiers = create_default_modifiers_for_team(team, modifiers)
+
     database = Database(timezone=team.timezone, week_start_day=team.week_start_day)
 
+    # Only internal users can access query_log
+    if (
+        (get_instance_region() == "EU" and team_id == EU_INSTANCE_TEAM_ID)
+        or (get_instance_region() == "US" and team_id == US_INSTANCE_TEAM_ID)
+        or settings.DEBUG
+    ):
+        database = InternalDatabase(timezone=team.timezone, week_start_day=team.week_start_day)
```

Comment:
Not sure if `InternalDatabase` is the correct naming here (nor whether subclassing `Database`. I was imagining you using `setattr(..)` to add the new tables to an existing `Database` object

Reply:
Issue with `setattr` was that these are class variables so if we set them once, then they'll persist on the class. Separate classes solves this issue